File: PP_02.01/DrivingSchoolWebAPI2/DrivingSchoolAPIModels/Enums.cs
namespace DrivingSchoolAPIModels
{
    /// <summary>
    /// Оценки от инструкторов ученикам
    /// </summary>
    public enum GradesByInstructorsToStudents : byte
    {
        /// <summary>
        /// Прогул
        /// </summary>
        Прогул = 1,
        Удовлетворительно,
        Хорошо,
        Отлично
    }
    /// <summary>
    /// Оценки от учеников инструкторам
    /// </summary>
    public enum GradesByStudentsToInstructors : byte 
    {
        Ужасно = 1, 
        Плохо, 
        Удовлетворительно, 
        Хорошо, 
        Отлично
    };
    /// <summary>
    /// Состояния занятий
    /// </summary>
    public enum ClassStatus : byte { Предстоит, Завершено, Отменено }
    public enum ClassType : byte { Inner, Outer }
}

File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/Enums.kt
package com.example.drivingschoolappandroidclient.api.models

/**
 * Оценки от инструкторов ученикам
 */
enum class GradesByInstructorsToStudents(val value: Byte) {
    Прогул(1),
    Удовлетворительно(2),
    Хорошо(3),
    Отлично(4);
    constructor(str: String) : this(str.toByte())
    companion object{
        fun ofValue(value: Byte) = GradesByInstructorsToStudents.values().first { it.value == value }
    }
}

/**
 * Оценки от учеников инструкторам
 */
enum class GradesByStudentsToInstructors(val value: Byte) {
    Ужасно(1),
    Плохо(2),
    Удовлетворительно(3),
    Хорошо(4),
    Отлично(5);
    constructor(str: String) : this(str.toByte())
    companion object{
        fun ofValue(value: Byte) = GradesByStudentsToInstructors.values().first { it.value == value }
    }
}

/**
 * Состояния занятий
 */
enum class ClassStatus(val value: Byte) {
    Предстоит(0),
    Завершено(1),
    Отменено(2);
    constructor(str: String) : this(str.toByte())
    companion object{
        fun ofValue(value: Byte) = ClassStatus.values().first { it.value == value }
    }
}

enum class ClassType(val value: Byte) {
    Inner(0),
    Outer(1);
    constructor(str: String) : this(str.toByte())
    companion object{
        fun ofValue(value: Byte) = ClassType.values().first { it.value == value }
    }
}
